File: Pyhon/1160.py
c = int(input())

for i in range(1,c+1):
    entrada = input().split(" ")
    pa = int(entrada[0])
    pb = int(entrada[1])
    g1 = float(entrada[2])
    g2 = float(entrada[3])

    anos = 0
    while pa <= pb:
        pa += int(pa * (g1 / 100))
        pb += int(pb * (g2 / 100))
        anos += 1
        if anos > 100:
            print("Mais de 1 seculo.")
            break

    if anos <= 100:
        print(anos,"anos.")

File: C++/1160.cpp
#include <iostream>

int main()
{
    long long n, PA, PB;
    double G1, G2;
    int anos, sec;

    std::cin >> n;

    for (int i = 0; i < n; i++)
    {
        std::cin >> PA;
        std::cin >> PB;
        std::cin >> G1;
        std::cin >> G2;

        anos = 0;
        sec = 0;

        while (PA <= PB)
        {
            PA += PA * G1 / 100;
            PB += PB * G2 / 100;

            anos++;
            if (anos > 100)
            {
                sec = 1;
                break;
            }
        }
        if (sec == 0)
            std::cout << anos << " anos.\n";
        else
            std::cout << "Mais de 1 seculo.\n";
    }
    return 0;
}